### PR TITLE
New Property `allowPanning`

### PR DESCRIPTION
**Pull request by escamoteur**
Added a new property to be able to disable panning while still keep the View interactive and added a safety check on PolylinePainter

### CODE REVIEW

**Comment by johnpryan** [comment]
It looks like there's still a few analysis errors:

```
20.30s$ ~/flutter/bin/cache/dart-sdk/bin/dartanalyzer .
Analyzing flutter_map...
  error • The named parameter 'colors' isn't defined. • example/lib/pages/animated_map_controller.dart:93:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/animated_map_controller.dart:103:30 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/home.dart:19:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/home.dart:30:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/home.dart:41:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/live_location.dart:108:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/map_controller.dart:51:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/map_controller.dart:61:30 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/marker_anchor.dart:54:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/marker_anchor.dart:64:30 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/on_tap.dart:51:13 • undefined_named_parameter
  error • The named parameter 'colors' isn't defined. • example/lib/pages/on_tap.dart:66:30 • undefined_named_parameter
12 errors found.
```

**Comment by escamoteur** [comment]
Hi @johnpryan I pulled in the latest changes and I now only see some warnings concnering Scaffold messenger.

**Comment by johnpryan** [comment]
What flutter channel are you using? The Travis check uses stable.

I think you need to run `dartfmt` to make the Travis check green.

**Comment by escamoteur** [comment]
Uups, yeah I m on master but I have dartfmt switched to automatic. Will do tomorrow Am 06.01.2021 22:38 schrieb John Ryan <notifications@github.com>:
What flutter channel are you using? The Travis check uses stable.
I think you need to run dartfmt to make the Travis check green.

**Comment by escamoteur** [comment]
yep, one file wasn't formated properly

File: lib/src/layer/polyline_layer.dart [hunk 1]
```diff
@@ -150,12 +150,15 @@ class PolylinePainter extends CustomPainter {
...
     }
 
-    final filterPaint = Paint()
-      ..color = polylineOpt.borderColor.withAlpha(255)
-      ..strokeWidth = polylineOpt.strokeWidth
-      ..strokeCap = StrokeCap.round
-      ..strokeJoin = StrokeJoin.round
-      ..blendMode = BlendMode.dstOut;
+    Paint filterPaint;
+    if (polylineOpt.borderColor != null) {
```
Comment (johnpryan) [inline lib/src/layer/polyline_layer.dart:154]: Is this change intentional?

File: lib/src/layer/polyline_layer.dart [hunk 2]
```diff
@@ -181,11 +184,13 @@ class PolylinePainter extends CustomPainter {
       canvas.restore();
     } else {
       paint.style = PaintingStyle.stroke;
-      filterPaint.style = PaintingStyle.stroke;
-      borderPaint?.style = PaintingStyle.stroke;
       canvas.saveLayer(rect, Paint());
       if (borderPaint != null) {
-        _paintLine(canvas, polylineOpt.offsets, borderPaint);
+        if (filterPaint != null) {
```
Comment (johnpryan) [inline lib/src/layer/polyline_layer.dart:189]: same here

Comment (escamoteur) [reply lib/src/layer/polyline_layer.dart:189]: That's the follow up from the above

File: lib/src/layer/polyline_layer.dart (hunk 1 above)
Comment (escamoteur) [reply lib/src/layer/polyline_layer.dart:154]: yeah, that's from an much older commit. I recall that I got null reference exceptions when borderColor wasn't set

Comment (escamoteur) [reply lib/src/layer/polyline_layer.dart:154]: Do you see any problem with that?

Comment (johnpryan) [reply lib/src/layer/polyline_layer.dart:154]: No problem, just harder to review :)
